pkg/node/provisioners/drive_provisioner.go

```diff
@@ -224,7 +224,15 @@ func (d *DriveProvisioner) GetVolumePath(vol *api.Volume) (string, error) {
 	}
 	volumeUUID, _ = util.GetVolumeUUID(volumeUUID)
 
-	partNum, err := d.partOps.SearchPartName(device, volumeUUID)
+	// For volumes that are in Published status, ignore partition re-read and
+	// get the partition number directly.
+	var partNum string
+	if vol.CSIStatus == apiV1.Published {
+		partNum, err = d.partOps.GetPartitionNameByUUID(device, volumeUUID)
+	} else {
+		partNum, err = d.partOps.SearchPartName(device, volumeUUID)
```

Comment:
We have a case when a volume in PUBLISHED state but not really mounted - after physical node reboot kubelet doesn't send UnStage and UnPublish requests and when node is back we receive NodeStage request for PUBLISHED volume (and have to bind/mount it again)
Could you please clarify a difference between `partprobe` and `blockdev` usage - why we have an error here? And could we use another tool for partition syncing but without udev calls?

Reply:
hi @safronovD,   I think the fix here will handle the case you described properly.  it  just does not re-read (resync) the kernel partition table again.  it still will get the part uuid directly and then do subsequent mount operations. 

for the difference b/w partprobe and blockdev,  I see blockdev will use the ioctl (BLKPART)  syscall call to do related operation,  while partprobe not use the same one and I found a link that said partprobe indeed did some improvment for mounted device. (https://bugs.freedesktop.org/show_bug.cgi?id=20813#:~:text=This%20is%20caused%20by%20the%20BLKRRPART%20ioctl%20failing,is%20really%20just%20a%20bug%2Ffeature%20in%20the%20kernel)

regarding the tool,  I do some research and found cfdisk/sfdisk,  but they all use the same api as blockdev(ioctl BLKPART)  and will get the same error.

Reply:
thanks @yurkov-anton for the review. 
do you mean skip blockdev error "device or resource busy" in underlying SearchPartName function ?
then it will need many code changes to pass the volume CSI status to the function, now the search PartName did not care about the volume,  just the uuid. 
another thing is "device or resource busy" is just the blockdev command stdout,  the err returned is "exit status -1",  so it is not easy to judge this.

Reply:
Can we check for "Block device is busy"  in a error returned by `SearchPartName`?
Skiping synch of part table can be cause undefined behavior when disk was changed or disconnected.

Reply:
Hi @yurkov-anton,  in current code,  if SearchPartName returns error,  then no partuuid is returned, and subsequent action such as mounting the volume could not be done.  that's the reason I do a replacement for only get partuuid if volume is already in published status.  you can check the code call chain as below,  if we did not skip the partition re-read in such scenario, then it will fall into such  
error loop as now the problem describeed, 
```
   NodeStageVolume
        GetVolumePath
            SearchPartName
                SyncPartitionTable
```
here why we skip partition resync(partition re-read to notify the kernel) , since volume was already in published status, that indicated the partition was already be notified before,  even the node was rebooted,  that partition would not change.  we just need to re-read the partition uuid, no need to notify the kernel,  right?

Reply:
Hi @yurkov-anton,  sorry to bother you again,  could you help review again? any suggestions for my above comments ?

Reply:
Hi @libzhang, sorry for the long answer.
I've told about ignore synch error only if message contain "Disk busy"  and volume is PUBLISHED, because this is expected error in this way. 
```
partName, err = SearchPartName
if err {
    if "Disk is busy" in err and vol.CSIStatus == apiV1.Published {
        partName, err = GetPartitionNameByUUID
   }
   if err {
         return err
   }
}
```
We have case that @safronovD was talking about:
1. physical node reboot
2. something wrong with disk or removed physically drive
3. VolumeCR still published and OS keep path to disk.
4. node is back and NodeStage send
5. CSI get part path by SearchPartName, but path is missing

I think can be another cases, which we do not take into account. We need to synch part table, if we can, before use part path.

Reply:
thanks @yurkov-anton.  I got your point.  will work on this way.

Reply:
Hi @yurkov-anton,  changed the code based on your comments,  may you help review ? thanks.